### PR TITLE
Use more general Kitodo directory mapping

### PR DESCRIPTION
**Pull request by Beacze**
1. Separate map to Kitodo URI on three different types of mapping:
- config
- data
- root (apache)
2. Handle cases when return value was ignored
3. Add MappingType enum
4. Order methods in FileService

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 1]
```diff
@@ -215,24 +248,35 @@ public URI renameFile(URI fileUri, String newFileName) throws IOException {
...
-                    count += getNumberOfFiles(aChildren);
-                } else {
-                    count += 1;
+            if (isDirectory(directory)) {
+                for (URI aChildren : children) {
+                    if (isDirectory(aChildren)) {
+                        count += getNumberOfFiles(aChildren);
+                    } else {
+                        count += 1;
+                    }
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: Both iterations through the list of URIs looks similar: why did you not move this counting to its own method? This would even reduce the level of nested `if` statements.

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 2]
```diff
@@ -191,7 +223,8 @@ public URI renameFile(URI fileUri, String newFileName) throws IOException {
                 }
                 millisWaited += SLEEP_INTERVAL_MILLIS;
             }
-        } while (!success && millisWaited < MAX_WAIT_MILLIS);
+        }
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: Is this format change indented or valid with the used format rules?

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 3]
```diff
@@ -241,15 +285,17 @@ public Integer getNumberOfFiles(URI directory) {
      */
     public Integer getNumberOfImageFiles(URI directory) {
         int count = 0;
-        if (isDirectory(directory)) {
-            /*
-             * die Images zählen
-             */
+        if (directory.isAbsolute()) {
+            if (isDirectory(directory)) {
+                count = getSubUris(Helper.imageNameFilter, directory).size();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: Why did you add the number of subdirectories as a start value for the number of image files? Looks a bit strange.

Like in method `getNumberOfFiles` both code lines of the `if`statement looks similar and could be separated.

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java (hunk 2 above)
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: Looks like lost code. A `do { … }` without `while ( … )` shouldn’t even compile.

Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: The corresponding `do` is in line 175 / 206.

Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: actually it is interesting case, auto formatter says:
` } while`
and checkstyle says
```
} 
while
```

Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: I like the 
} while
better. But i don't care, we should just decide for one!

Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: so I will adjust checkstyle to accept this :P

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java (hunk 3 above)
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: I didn't add it, I was not changing logic of the method:
https://github.com/Beacze/kitodo-production/commit/e0b8a5e341e691cb7bb2a39a9f3088ea480bd871

**Comment by henning-gerhardt** [comment]
@Beacze : your changes looks good.
@Kathrin-Huber: do you want to review this pull request?

**Comment by Kathrin-Huber** [comment]
I already reviewed it! :)
